app/models/user.rb

```diff
@@ -14,4 +14,15 @@ def self.from_twitter(auth)
       user.password = SecureRandom.hex
     end
   end
+
+  def update_with_twitter(auth)
+    self.name = auth.info.nickname if name.nil? || name.blank?
+    self.uid = auth.uid
+
+    save!
+  end
+
+  def linked_with_twitter?
+    !(uid.nil? || uid.blank?)
+  end
 end
```

Comment:
linked_with_twitter? determines whether the user account has been linked with Twitter. It looks for the presence of uid as the underlying implementation. The implementation could change here without affecting callers of this method.